lib/streamlit/elements/vega_charts.py

```diff
@@ -1088,15 +1263,30 @@ def _altair_chart(
         altair_chart: alt.Chart,
         use_container_width: bool = False,
         theme: Literal["streamlit"] | None = "streamlit",
+        key: Key | None = None,
+        on_select: Literal["rerun", "ignore"] | WidgetCallback = "ignore",
         add_rows_metadata: AddRowsMetadata | None = None,
-    ) -> DeltaGenerator:
-        """Internal method to enqueue a vega-lite chart element based on an Altair chart."""
+    ) -> DeltaGenerator | VegaLiteState:
+        """Internal method to enqueue a vega-lite chart element based on an Altair chart.
+
+        See the `altair_chart` method docstring for more information.
+        """
+
+        if type_util.is_altair_version_less_than("5.0.0") and on_select != "ignore":
+            raise StreamlitAPIException(
+                "Streamlit does not support selections with Altair 4.x. Please upgrade to Version 5. "
+                "If you would like to use Altair 4.x with selections, please upvote "
+                "this [Github issue](https://github.com/streamlit/streamlit/issues/8516)."
+            )
```

Comment:
We should add to the docstring that this exception can be raised by the function

Reply:
But its added in the docstring or? 

> Check if the spec has any selections defined. If not, raise an exception.